src/components/VmActions/index.js

```diff
@@ -134,15 +136,40 @@ class VmActions extends React.Component {
 
     const vncConsole = vm.get('consoles').find(c => c.get('protocol') === 'vnc')
     const hasRdp = isWindows(vm.getIn(['os', 'type']))
-    const consoles = vm.get('consoles').map((c) => ({
-      priority: 0,
-      shortTitle: msg[c.get('protocol') + 'Console'](),
-      icon: <Icon name='external-link' />,
-      id: `${idPrefix}-button-console-${c.get('protocol')}`,
-      confirmation: (
-        <ConsoleConfirmationModal consoleId={c.get('id')} vm={vm} />
-      ),
-    })).toJS()
+    let consoles = []
+    if (isOnCard) {
+      consoles.push({
+        priority: 1,
+        actionDisabled: isPool || !canConsole(status) || vm.getIn(['actionInProgress', 'getConsole']),
+        shortTitle: hasRdp ? msg.nonRdpConsole() : msg.console(),
+        onClick: (e) => { e.preventDefault() },
+        className: 'btn btn-success',
+        id: `${idPrefix}-button-Consoles`,
+        confirmation: (<AutoSelectConsole vm={vm} id={`${vm.get('id')}-console-selector`} />),
+      })
+    } else {
+      consoles = vm.get('consoles').map((c) => ({
+        priority: 0,
+        shortTitle: msg[c.get('protocol') + 'Console'](),
+        icon: <Icon name='external-link' />,
+        id: `${idPrefix}-button-console-${c.get('protocol')}`,
+        confirmation: (
+          <ConsoleConfirmationModal consoleId={c.get('id')} vm={vm} />
+        ),
+      })).toJS()
+
+      if (vncConsole) {
+        consoles.push({
+          priority: 0,
+          shortTitle: msg.vncConsoleBrowser(),
+          actionDisabled: config.get('websocket') === null,
+          id: `${idPrefix}-button-console-browser`,
+          confirmation: (
+            <ConsoleConfirmationModal isNoVNC consoleId={vncConsole.get('id')} vm={vm} />
```

Comment:
It's still there:)

Reply:
`isNoVNC` is the correct attribute to be there in this case.  It tells the confirmation modal to open VNC in the "noVnc" in-browser mode.  The map through the consoles above will generate the confirmation modal to open VNC in the "console.vv" style.

Reply:
@sjd78  @bamsalem 
OK.  I was confused by the `isNoVnc` prop introduced in AutoSelectConsole which was very similar to `isNoVNC` prop used in ConsoleConfirmationModal. Not passing this property to  ConsoleConfirmationModal seemed a typo to me.
I still think that launching VNC console with `isNoVNC == true` is confusing. Perhaps a different name would be better explain the purpose? However this refactoring is out of scope of this patch so let's leave it as it is.